src/onegov/search/indexer.py

```diff
@@ -307,84 +311,106 @@ def delete(self, task):
-        self.psql_client = engine
-        self.queue = queue
-        self.session = session
-        self.failed_task = None
-
-    def process(self):
-        """ Processes the queue until it is empty or blocking for 1 second.
-        :returns: The number of successfully processed items
-        """
-
-        def action(task):
-            if task['action'] == 'index_obj':
-                retval = self.index_obj(task['obj'])
-            elif task['action'] == 'delete_obj':
-                retval = self.delete_obj(task['obj'])
-            else:
-                raise NotImplementedError
 
-            return retval
-
-        processed = 0
-        while True:
-            try:
-                task = (self.failed_task
-                        or self.queue.get(block=True, timeout=1))
-                self.failed_task = None
+    def __init__(self, mappings, queue, es_client, engine, session):
+        super().__init__(mappings, queue, es_client, hostname=None)
 
-                if action(task):
-                    processed += 1
-                    self.queue.task_done()
-
-            except Empty:
-                break
-
-            except Exception as ex:
-                print(f'Error PostgresIndexer::process Error processing '
-                      f'task {task} ex: {ex}')
+        self.engine = engine
+        self.session = session
 
-        return processed
+        self.language_mapping = {
+            'de': 'german',
+            'fr': 'french',
+            'it': 'italian',
+            'en': 'english',
+        }
+        self.index_issue_count = 0
 
-    def index_obj(self, obj):
-        """ Update the 'fts_idx_data' column of the given object. The index
-        is automatically updated as it depends on the 'fts_idx_data' column.
+    def index(self, task):
+        """ Update the 'fts_idx' column (full text search index) of the given
+        object.
         """
 
         try:
-            session = self.session()
-            if hasattr(obj, 'id'):
-                obj = session.query(obj.__class__).filter_by(
-                    id=obj.id).first()
-            elif hasattr(obj, 'name'):
-                obj = session.query(obj.__class__).filter_by(
-                    name=obj.name).first()
-            else:
-                raise NotImplementedError
-
-            data = obj.gather_fts_index_data()
-            if obj.fts_idx_data != data:
-                # this check prevents unnecessary updates causing 'update'
-                # events triggering reindexing the object leading to an
-                # endless loop
-                obj.fts_idx_data = data
+            data = []
+            languages = ['simple']
+
+            properties = task['properties']
+            if task['language'] in self.language_mapping:
+                languages.append(self.language_mapping[task['language']])
+
+            data = self.get_data_from_properties(data, properties)
```

Comment:
Don't try to recompute the data, we already have the data embedded in the task:
```suggestion
            data = task['properties'].copy()
            # TODO: handle suggestions
            del data['es_suggestion']
```

Reply:
I adapted it to remove all keys starting with 'es_'